private/PROPOSAL/crossection/parametrization/PhotoQ2Integration.cxx

```diff
@@ -616,4 +618,225 @@ double PhotoRenoSarcevicSu::FunctionToQ2Integral(double energy, double v, double
+        c_reggeon = c1_reggeon + c2_reggeon * std::pow(t, c3_reggeon);
+    } else {
+        a_reggeon = a1_reggeon;
+        c_reggeon = c1_reggeon;
+    }
+
+    // parameter, that decrease with Q^2
+    // eq. 24
+    // f'(t) = f_1 + (f_1 - f_2) (\frac{1}{1 + t^{f_3}} - 1)
+    double a_pomeron = a1_pomeron + (a1_pomeron - a2_pomeron) * (1 / (1 + std::pow(t, a3_pomeron)) - 1);
+    double c_pomeron = c1_pomeron + (c1_pomeron - c2_pomeron) * (1 / (1 + std::pow(t, c3_pomeron)) - 1);
+
+    // invariant mass of nucleus and virtual photon
+    // W^2 = (p + q)^2 = M^2 + 2MEv - Q^2
+    double W2 = mass_nucleus * mass_nucleus + 2 * mass_nucleus * energy * v - Q2;
+
+    // Relation between structure function of proton and structure function of neutron
+    // from the BCDMS Collaboration
+    // Phys. Lett. B 237 (1990), 599
+    // eq. 4
+    // P(x) = 1 - 1.85x + 2.45x^2 - 2.35x^3 + x^4
+    double relation_proton_neutron = bjorken_x * bjorken_x;
+    relation_proton_neutron        = 1 - 1.85 * bjorken_x + 2.45 * relation_proton_neutron -
+                              2.35 * relation_proton_neutron * bjorken_x +
+                              relation_proton_neutron * relation_proton_neutron;
+
+    // eq. 17 and 18
+    // x_i = \frac{Q^2 + m_i}{Q^2 + m_i + W^2 - M^2}
+    double bjorken_x_pomeron = (Q2 + mass_pomeron) / (Q2 + mass_pomeron + W2 - mass_nucleus * mass_nucleus);
+    double bjorken_x_reggeon = (Q2 + mass_reggeon) / (Q2 + mass_reggeon + W2 - mass_nucleus * mass_nucleus);
+    // eq. 20 and 21
+    // F_{2, i}(x, Q^2) = c_i(t) x^{a_i(t)} (1 - x)^{b_i(t)}
+    double pomeron_contribution = c_pomeron * std::pow(bjorken_x_pomeron, a_pomeron) * std::pow(1 - bjorken_x, b_pomeron);
+    double reggeon_controbution = c_reggeon * std::pow(bjorken_x_reggeon, a_reggeon) * std::pow(1 - bjorken_x, b_reggeon);
+    // ALLM97 eq. 1
+    // F_{2, proton}(x, Q^2) = \frac{Q^2}{Q^2 + m_0^2} (F_{2, Pomeron} + F_{2, Reggeon})
+    double structure_function_proton = Q2 / (Q2 + mass_photon_eff) * (pomeron_contribution + reggeon_controbution);
+    // structure function from Dutta et al
+    // Phys. Rev. D 63 (2001), 094020
+    // eq. 3.11
+    // F_{2, nucleus} = G(x) (Z + (A - Z)P(x)) F_{2, Proton}
+    double structure_function_nucleus =
+        structure_function_proton * shadow_effect_->CalculateShadowEffect(component, bjorken_x, v * energy) *
+        (component.GetNucCharge() + (component.GetAtomicNum() - component.GetNucCharge()) * relation_proton_neutron);
+
+    // differential cross section from Dutta et al.
+    // Phys. Rev. D 63 (2001), 094020
+    // eq. 3.11
+    // eq. 3.4
+    // \frac{d^2 \sigma}{dQ^2 dx} = \frac{4\pi\alpha^2}{Q^4} \frac{F_{2, Proton}}{v}
+    //     ( 1 - v + \frac{M x v}{2E}
+    //     - \frac{v^2}{2} (1 - \frac{2m_{particle}}{Q^2})
+    //          \frac{1 + \frac{4 M^2 x^2}{Q^2}}{1 + R})
+    double result = ME * RE / Q2;
+    result *= result * 4 * PI * structure_function_nucleus / v *
```

Comment:
Is this really correct? `result *= result ... ` look looks suspicious.

Reply:
yes, that's quite often the way how things are squared in PROPOSAL.
It's probably not code-reader friendly

Reply:
Maybe worth taking a look to implement a `pow(double, int)` ?

https://stackoverflow.com/a/13771544/3838691

Reply:
Actually, all current compilers just replace `pow(x, 2)` with `x * x`: https://godbolt.org/z/a3c5sY

Reply:
So the above is only beneficial for integer powers > 2